fil_fungible_token/src/token/mod.rs

```diff
@@ -294,6 +304,32 @@ where
         Ok(())
     }
 
+    /// Sets the allowance to a specified amount, returning the old allowance
+    pub fn set_allowance(
+        &mut self,
+        owner: &Address,
+        operator: &Address,
+        amount: &TokenAmount,
+    ) -> Result<TokenAmount> {
+        if amount.is_negative() {
+            return Err(TokenError::InvalidNegative { name: "allowance", amount: amount.clone() });
```

Comment:
The state method already does this.

Reply:
I've removed it from the library level and let the state level handle this